onnx/common/tensor.h

```diff
@@ -27,6 +29,115 @@ struct Tensor final {
   bool is_raw_data_;
   std::string raw_data_;
 
+  template<typename T>
+  static void add_nums(T* x, const T* y) {
+    *x += *y;
+  }
+
+  template<typename T>
+  static void sub_nums(T* x, const T* y) {
+    *x -= *y;
+  }
+
+  template<typename T>
+  static void mult_nums(T* x, const T* y) {
+    *x *= *y;
+  }
+
+  template<typename T>
+  static void divide_nums(T* x, const T* y) {
+    *x /= *y;
+  }
+
+  template<typename T>
+  static void sqrt_num(T* x) {
+    *x = (T) std::sqrt((double) *x);
+  }
+
+
+  template<typename T, int64_t block_size>
+  void bin_func(void (*f)(T*, const T*), Tensor& a, std::vector<T>& T_data_, std::vector<T>& a_T_data_) {
```

Comment:
I think you only need `a_T_data` (or `a`) here? Also they should be const

Reply:
made const, but you need both because you don't know where to look for the data if the tensor is stored as raw data - unless you do another switch on the types